src/layouts/PLayoutResizable/PLayoutResizable.vue

```diff
@@ -0,0 +1,174 @@
+      <slot name="aside" v-bind="{ collapsed, expand, collapse, toggle }" />
+    </aside>
+
+    <div
+      v-if="slots.default && slots.aside"
+      class="p-layout-resizable__divider"
+      :class="classes.divider"
+      @mousedown="start"
+      @mouseup="handleAsideClick"
+    />
+
+    <main v-if="slots.default" class="p-layout-resizable__main" :class="classes.main">
+      <slot v-bind="{ collapsed, expand, collapse, toggle }" />
+    </main>
+  </section>
+</template>
+
+<script lang="ts" setup>
+  import { computed, ref, useSlots } from 'vue'
+
+  const props = defineProps<{
+    collapsePoint?: number,
+    disabled?: boolean,
+  }>()
+
+  const slots = useSlots()
+
+  const container = ref<HTMLDivElement>()
+  const aside = ref<HTMLDivElement>()
+
+  const dragging = ref(false)
+  const collapsed = ref(false)
+
+  const classes = computed(() => ({
+    aside: {
+      'p-layout-resizable--events-none': dragging.value,
+      'p-layout-resizable__aside--collapsed': collapsed.value,
+    },
+    divider: {
+      'p-layout-resizable__divider--resizable': !props.disabled && !dragging.value,
+      'p-layout-resizable__divider--collapsed': collapsed.value,
+    },
+    main: {
+      'p-layout-resizable--events-none': dragging.value,
+    },
+  }))
+
+  let asideLeft = 0
+  const drag = (event: MouseEvent): void => {
+    if (props.disabled || !container.value || !aside.value) {
+      return
+    }
+
+    if (props.collapsePoint) {
+      collapsed.value = event.clientX - asideLeft < props.collapsePoint
+    }
+
+    aside.value.style.width = `${event.clientX - asideLeft}px`
+  }
+
+  const start = (): void => {
+    if (props.disabled || !container.value || !aside.value) {
+      return
+    }
+
+    const asideBounds = aside.value.getBoundingClientRect()
+    asideLeft = asideBounds.left
+    document.body.style.cursor = 'col-resize'
+    dragging.value = true
+    document.body.style.userSelect = 'none'
+    window.addEventListener('mouseup', stop)
+    window.addEventListener('mousemove', drag)
+  }
+
+  const stop = (): void => {
+    window.removeEventListener('mousemove', drag)
+    window.removeEventListener('mouseup', stop)
+    document.body.style.userSelect = 'auto'
+    dragging.value = false
+    document.body.style.removeProperty('cursor')
+  }
+
+  const handleAsideClick = (): void => {
+    if (!props.disabled && collapsed.value) {
+      expand()
+    }
+  }
+
+  const expand = (): void => {
+    collapsed.value = false
+  }
+
+  const collapse = (): void => {
+    collapsed.value = true
+  }
+
+  const toggle = (): void => {
+    collapsed.value = !collapsed.value
+  }
+</script>
+
+<style>
+.p-layout-resizable {
+  --aside-width: var(--p-layout-resizable-aside-width, min-content);
+  --aside-max-width: var(--p-layout-resizable-aside-max-width, theme(width.80));
+  --aside-min-width: var(--p-layout-resizable-aside-min-width, theme(width.20));
+}
+
+.p-layout-resizable { @apply
+  flex
```

Comment:
I'm a little concerned about using flex here. I've just been burned by flex "blowout" a lot. But I'm also not sure if you could do this with grid. 

I think at the very least we might need a `min-width: 0` on the `__main` element.

Reply:
Yeah makes sense - I added that. If that becomes a problem I'll revisit this but for now it works nicely without